boot/bootutil/src/bootutil_public.c

```diff
@@ -316,14 +316,30 @@ int
 boot_write_magic(const struct flash_area *fap)
 {
     uint32_t off;
+    uint32_t pad_off;
     int rc;
+    uint8_t magic[BOOT_MAGIC_ALIGN_SIZE];
+    uint8_t erased_val;   
 
     off = boot_magic_off(fap);
+    /* image_trailer structure was modified with additional padding such that 
+     * the pad+magic ends up in a flash minimum write region. The address 
+     * returned by boot_magic_off() is the start of magic which is not the 
+     * start of the flash write boundary and thus writes to the magic will fail. 
+     * To account for this change, write to magic is first padded with 0xFF 
+     * before writing to the trailer. */
+    pad_off = off & ~(BOOT_MAX_ALIGN - 1);
+    
+    erased_val = flash_area_erased_val(fap);
+
+    memset(&magic[0], erased_val, sizeof(magic));
```

Comment:
set size should be (BOOT_MAGIC_ALIGN_SIZE - BOOT_MAGIC_SZ)

Reply:
Clang static analyzer emits a warning due to the possibility of `(BOOT_MAGIC_ALIGN_SIZE - BOOT_MAGIC_SZ)` resulting in a call to `memset` with 0 size.
Since filling the `magic` buffer with the `erased_val` pattern is not wrong either, I believe there is no problem leaving it like this. Do you agree?

Reply:
Both C99 and C11 allow to have 0 length here for size_t type parameter of <string.h> library functions.
But if Clang complains I agree that we can just do what is done here.